luigi/contrib/beam_dataflow.py

```diff
@@ -0,0 +1,451 @@
+
+    runner                  # PipelineRunner implementation for your Beam job.
+                              Default: DirectRunner
+    num_workers             # The number of workers to start the task with
+                              Default: Determined by Dataflow service
+    autoscaling_algorithm   # The Autoscaling mode for the Dataflow job
+                              Default: `THROUGHPUT_BASED`
+    max_num_workers         # Used if the autoscaling is enabled
+                              Default: Determined by Dataflow service
+    network                 # Network in GCE to be used for launching workers
+                              Default: a network named "default"
+    subnetwork              # Subnetwork in GCE to be used for launching workers
+                              Default: Determined by Dataflow service
+    disk_size_gb            # Remote worker disk size. Minimum value is 30GB
+                              Default: set to 0 to use GCP project default
+    worker_machine_type     # Machine type to create Dataflow worker VMs
+                              Default: Determined by Dataflow service
+    job_name                # Custom job name, must be unique across project's
+                              active jobs
+    worker_disk_type        # Specify SSD for local disk or defaults to hard
+                              disk as a full URL of disk type resource
+                              Default: Determined by Dataflow service.
+    service_account         # Service account of Dataflow VMs/workers
+                              Default: active GCE service account
+    region                  # Region to deploy Dataflow job to
+                              Default: us-central1
+    zone                    # Availability zone for launching workers instances
+                              Default: an available zone in the specified region
+    staging_location        # Cloud Storage bucket for Dataflow to stage binary
+                              files
+                              Default: the value of temp_location
+    gcp_temp_location       # Cloud Storage path for Dataflow to stage temporary
+                              files
+                              Default: the value of temp_location
+    labels                  # Custom GCP labels attached to the Dataflow job
+                              Default: nothing
+    """
+
+    project = None
+    runner = None
+    temp_location = None
+    staging_location = None
+    gcp_temp_location = None
+    num_workers = None
+    autoscaling_algorithm = None
+    max_num_workers = None
+    network = None
+    subnetwork = None
+    disk_size_gb = None
+    worker_machine_type = None
+    job_name = None
+    worker_disk_type = None
+    service_account = None
+    zone = None
+    region = None
+    labels = {}
+
+    cmd_line_runner = _CmdLineRunner
+    dataflow_params = None
+    output_uris = None
```

Comment:
What is this supposed to be used for?

Reply:
cc @daikeshi, used in implementations of the post-run callbacks.